source/concepts.tex

```diff
@@ -816,64 +812,143 @@
-The \libconcept{boolean} concept specifies the requirements on a type that is
-usable as a truth value.
+The exposition-only \exposconcept{boolean-testable} concept
+specifies the requirements on expressions
+that are convertible to \tcode{bool} and
+for which the logical operators
+(\ref{expr.log.and}, \ref{expr.log.or}, \ref{expr.unary.op})
+have the conventional semantics.
 
 \begin{itemdecl}
-template<class B>
-  concept @\deflibconcept{boolean}@ =
-    movable<remove_cvref_t<B>> &&       // (see \ref{concepts.object})
-    requires(const remove_reference_t<B>& b1,
-             const remove_reference_t<B>& b2, const bool a) {
-      { b1 } -> convertible_to<bool>;
-      { !b1 } -> convertible_to<bool>;
-      { b1 && b2 } -> same_as<bool>;
-      { b1 &&  a } -> same_as<bool>;
-      {  a && b2 } -> same_as<bool>;
-      { b1 || b2 } -> same_as<bool>;
-      { b1 ||  a } -> same_as<bool>;
-      {  a || b2 } -> same_as<bool>;
-      { b1 == b2 } -> convertible_to<bool>;
-      { b1 ==  a } -> convertible_to<bool>;
-      {  a == b2 } -> convertible_to<bool>;
-      { b1 != b2 } -> convertible_to<bool>;
-      { b1 !=  a } -> convertible_to<bool>;
-      {  a != b2 } -> convertible_to<bool>;
-    };
+template<class T>
+  concept @\defexposconcept{boolean-testable-impl}@ = convertible_to<T, bool>;  // \expos
 \end{itemdecl}
 
 \pnum
-Given a type \tcode{B}, let \tcode{b1} and \tcode{b2} be
-lvalues of type \tcode{const remove_reference_t<B>}.
-\tcode{B} models \libconcept{boolean} only if
+Let \tcode{e} be an expression such that
+\tcode{decltype((e))} is \tcode{T}.
+\tcode{T} models \exposconcept{boolean-testable-impl} only if:
 
 \begin{itemize}
-\item \tcode{bool(b1) == !bool(!b1)}.
-\item \tcode{(b1 \&\& b2)}, \tcode{(b1 \&\& bool(b2))}, and
-      \tcode{(bool(b1) \&\& b2)} are all equal to
-      \tcode{(bool(b1) \&\& bool(b2))}, and have the same short-circuit
-      evaluation.
-\item \tcode{(b1 || b2)}, \tcode{(b1 || bool(b2))}, and
-      \tcode{(bool(b1) || b2)} are all equal to
-      \tcode{(bool(b1) || bool(b2))}, and have the same short-circuit
-      evaluation.
-\item \tcode{bool(b1 == b2)}, \tcode{bool(b1 == bool(b2))}, and
-      \tcode{bool(bool(b1) == b2)} are all equal to
-      \tcode{(bool(b1) == bool(b2))}.
-\item \tcode{bool(b1 != b2)}, \tcode{bool(b1 != bool(b2))}, and
-      \tcode{bool(bool(b1) != b2)} are all equal to
-      \tcode{(bool(b1) != bool(b2))}.
+\item
+either \tcode{remove_cvref_t<T>} is not a class type, or
+name lookup for the names \tcode{operator\&\&} and \tcode{operator||}
+within the scope of \tcode{remove_cvref_t<T>}
+as if by class member access lookup\iref{class.member.lookup}
+results in an empty declaration set; and
+
+\item
+name lookup for the names \tcode{operator\&\&} and \tcode{operator||} in
+the associated namespaces and
+entities of \tcode{T}\iref{basic.lookup.argdep}
+finds no disqualifying declaration (defined below).
 \end{itemize}
 
+\pnum
+A \defnadj{disqualifying}{parameter}
```

Comment:
This is a local definition so I think the prevailing practice is to not index?

Reply:
This is a bit of a grey area. This seems convoluted and core-y enough that index entries and italics seem appropriate.